[JSC] Import Test262 (`8a8a7af1b9a01eb3665cedaa62131d5c1d2304de`)

Gumichocopengin8 opened the pull request:
#### 5e9a25f18c2c7650aaca46ff0108b486fc99b536
<pre>
[JSC] Import Test262 (`8a8a7af1b9a01eb3665cedaa62131d5c1d2304de`)
<a href="https://bugs.webkit.org/show_bug.cgi?id=299697">https://bugs.webkit.org/show_bug.cgi?id=299697</a>

Reviewed by Sosuke Suzuki.

This patch imports test262 [1].

[1]: <a href="https://github.com/tc39/test262/commit/8a8a7af1b9a01eb3665cedaa62131d5c1d2304de">https://github.com/tc39/test262/commit/8a8a7af1b9a01eb3665cedaa62131d5c1d2304de</a>

Canonical link: <a href="https://commits.webkit.org/300794@main">https://commits.webkit.org/300794@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a042d10e5bddfb89184e9086e31a9042ea3d8970

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123455 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/43170 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33866 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/130173 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75592 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43893 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51764 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93884 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62294 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126408 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34975 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110450 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74473 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33946 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73690 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/115614 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104690 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28833 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132892 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/121987 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50405 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38369 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102336 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50781 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106672 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102188 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/26071 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47545 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25770 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/47210 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50260 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/152341 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49733 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/38959 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/53081 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51409 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d832a7887ec5103288e7e258d979e07dafb552a1. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51465

sosukesuzuki commented on JSTests/test262/expectations.yaml:
```diff
@@ -1,6 +1,48 @@
 ---
 test/annexB/language/function-code/block-decl-func-skip-arguments.js:
   default: 'Test262Error: Expected SameValue(«"function arguments() {}"», «"[object Arguments]"») to be true'
+test/built-ins/ArrayBuffer/prototype/resize/this-is-immutable-arraybuffer-object.js:
```
I think we should exclude `immutable-array-buffer` because of we don't implement it yet

Gumichocopengin8 replied:
Updated

Gumichocopengin8 replied:
Because `JSTests/test262/test/built-ins/ArrayBuffer/prototype/slice/species-returns-immutable-arraybuffer.js` should be included.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a042d10e5bddfb89184e9086e31a9042ea3d8970. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51465

sosukesuzuki reviewed (approved):
r=me

Gumichocopengin8 committed:
[JSC] Import Test262 (`8a8a7af1b9a01eb3665cedaa62131d5c1d2304de`)
https://bugs.webkit.org/show_bug.cgi?id=299697

Reviewed by Sosuke Suzuki.

This patch imports test262 [1].

[1]: https://github.com/tc39/test262/commit/8a8a7af1b9a01eb3665cedaa62131d5c1d2304de

Canonical link: https://commits.webkit.org/300794@main
